pkgs/top-level/java-packages.nix

```diff
@@ -226,7 +226,7 @@ in {
       ../development/compilers/openjdk/22.nix
       ../development/compilers/zulu/22.nix
       {
-        openjdk22-bootstrap = temurin-bin.jdk-21;
+        jdk-bootstrap = temurin-bin.jdk-21;
```

Comment:
Is there a reason why this was changed? It breaks the pattern with all of the other JDK builders, so it seems a bit out of place to change it in just 1 place. Same with the other references to the bootstrap JDK.

Reply:
Probably I should have done it on a following PR. 
The main reason is to unify the arguments to `mkOpenJdk` and create a more generic function in the future. There is a lot of duplication between the 22.nix, 21.nix, etc. files. 

It also makes it more consistent with the openjfx argument. In my opinion, we should do this:

```nix
    openjdk22 = mkOpenjdk
      {
        jdk-bootstrap = temurin-bin.jdk-21;
        openjfx = openjfx22;
      };

```

or this:

```nix
    openjdk22 = mkOpenjdk
      {
        openjdk22-bootstrap = temurin-bin.jdk-21;
        openjdk22-openjfx = openjfx22;
      };
```

Reply:
@jlesquembre regarding this change `jdk-bootstrap` rename, a quick `git grep openjdk22-bootstrap` showed me this:

```
pkgs/development/compilers/openjdk/22.nix:, openjdk22-bootstrap
pkgs/development/compilers/openjdk/22.nix:  openjdk-bootstrap = openjdk22-bootstrap.override { gtkSupport = !headless; };
```

So, won't it break what's in `22.nix` and `21.nix`?

Reply:
I just undid that change